src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -263,6 +263,19 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
+							if (getParser().getCurrentStructure() != null) {
+								if (t.getUsableStructures().length > 0 && !CollectionUtils.contains(t.getUsableStructures(), getParser().getCurrentStructure().getClass())) {
```

Comment:
Would this handle superclasses? For example, if I only wanted my syntax to be used in some sort of abstract section class (ex: EffectSection)

Reply:
I don't think so, but i dont really see a way this could be done cleanly, as it only stores the base class?

Reply:
what I mean is if my method returns `EffectSection.class`, does the method here perform a simple `==` or like `requiredSection.isAssignableFrom(currentSection)`

Reply:
you could use isAssignableFrom to check if it's a subclass

Reply:
ah i understand, ill work on this now